0000-notes-on-oer-encoding/0000-notes-on-oer-encoding.md

```diff
@@ -0,0 +1,285 @@
+
+| Type | Size | Range |
+| :--- | :--- | :--- |
+| UInt8 | 1 byte | 0 .. 255 (2^8 - 1) |
+| UInt16 | 2 bytes | 0 .. 65535 (2^16 - 1) |
+| UInt32 | 4 bytes | 0 .. 4294967295 (2^32 - 1) |
+| UInt64 | 8 bytes | 0 .. 18446744073709551615 (2^64 - 1) |
+| UInt128 | 16 bytes | 0 .. 2^128 - 1 |
+| UInt160 | 20 bytes | 0 .. 2^160 - 1 |
+| UInt192 | 24 bytes | 0 .. 2^192 - 1 |
+| UInt224 | 28 bytes | 0 .. 2^224 - 1 |
+| UInt256 | 32 bytes | 0 .. 2^256 - 1 |
+| UInt384 | 48 bytes | 0 .. 2^384 - 1 |
+| UInt512 | 64 bytes | 0 .. 2^512 - 1 |
+
+For UInt64 and smaller, implementations SHOULD use internal methods that can serialize/deserialize any unsigned integer of size 1-8 bytes. This can also be used to decode the [length determinant](#length-determinant).
+
+Unsigned integers larger than 8 bytes MAY be represented as a byte string.
+
+#### Examples
+
+| Type | Bytes | Decoded value |
+| :--- | :--- | :--- |
+| UInt8 | `00` | `0` |
+| UInt16 | `1234` | `4660` |
+| UInt32 | `ABABABAB` | `2880154539` |
+| UInt64 | `AC01055A 1DEBAC1E` | `12394193534107495454` |
+| UInt256 | `FF713A73 8B32F2D3 29898CD9 7A42D75A`<br>`86D9E59E B3928E7B 7BFAADF4 A4689459` | `byte[32]` |
+| UInt512 | `37DA42AC 9C322C80 E5D7FD75 112CBEAD`<br>`B0B9FD10 E27A68FE 2DA16BE9 DB0BC10D`<br>`76EC90B0 BB136B13 EF033692 53119203`<br>`21B47236 C42FB4D1 A4DC52B6 DD0556E2` | `byte[64]` |
+
+### Fixed-length signed integers
+
+> Not currently used
+
+All fixed-length signed integers are encoded in big-endian, least-significant bit last. Negative values are encoded using 2's complement binary encoding.
+
+| Type | Size | Range |
+| :--- | :--- | :--- |
+| Int8 | 1 byte | -128 .. 127 |
+| Int16 | 2 bytes | -32768 .. 32767 |
+| Int32 | 4 bytes | -2147483648 .. 2147483647 |
+| Int64 | 8 bytes | -9223372036854775808 .. 9223372036854775807 |
+
+#### Examples
+
+| Type | Bytes | Decoded value |
+| :--- | :--- | :--- |
+| Int8 | `00` | `0` |
+| Int8 | `7F` | `127` |
+| Int8 | `80` | `-1` |
+| Int8 | `FF` | `-128` |
+| Int16 | `7FFF` | `32767` |
+| Int16 | `FC00` | `-1024` |
+| Int16 | `CFC7` | `-12345` |
+| Int32 | `0C00F5C9` | `201389513` |
+| Int32 | `F204BA10` | `-234571248` |
+| Int64 | `7FFFFFFF FFFFFFFF` | `9223372036854775807` |
+| Int64 | `EF68FE12 0BC51AD7` | `-8027945689248242392` |
+
+### Fixed-length floating point numbers
+
+> Not currently used
+
+Floating point values in Interledger are encoded using IEEE 754 binary floating point.
+
+| Type | Size | IEEE 754 name |
+| :--- | :--- | :--- |
+| Float32 | 4 bytes | binary32 |
+| Float64 | 8 bytes | binary64 |
+
+#### Examples
+
+| Type | Bytes | Decoded value |
+| :--- | :--- | :--- |
+| Float32 | `3F8FCD36` | `1.12345` (approximately) |
+| Float64 | `3FF1F9A6B50B0F28` | `1.12345` (approximately) |
+
+### Fixed-length octet string
+
+> Used in: ILP, BTP
+
+A fixed length octet string is encoded as itself with no added prefix or suffix.
+
+### Length determinant
```

Comment:
Based on https://github.com/interledger/rfcs/pull/387#discussion_r167688168, this doc should also include a mention of open types and where they're used in ILP